.github/workflows/release2.yaml

```diff
@@ -1,32 +1,59 @@
-name: release
+name: Build release
+
+# This is triggered by merging a PR from a branch named release-*. This workflow builds and publishes the release. Then it changes
+# the version to *-SNAPSHOT in various Java-related files.
 
 on:
-  push:
-    tags:
-    - '*'
+  pull_request:
+    types:
+    - closed
```

Comment:
This sounds a little scary. The comment above says the branch needs the name "release-", but where is that enforced? And does `closed` mean just merged, or if the PR is closed for any reason (perhaps that's covered by the conditional on line 13?)

Reply:
Thanks for catching that. I just fixed it.